Use `value_fn` with args of ComposedParameter (#228)
zfit/zfit: CHANGELOG.rst

 - SumPDF changed in the behavior. Read above in the Major Features and Improvement.
 - Integrals of extended PDFs are not extended anymore.

Depreceations
-------------
 - `ComposedParameter` takes now `params` instead of `dependents` as argument, it acts now as
   the arguments to the `value_fn`.

> 

Reviewer: I think we should state more explicitly that value_fn now needs to take explicit arguments instead of referencing the locals() and globals() dictionaries and implicate that this will force the user to change their value_fn if they want to stay compatible in the future.

Depreceations
-------------
 - `ComposedParameter` takes now `params` instead of `dependents` as argument, it acts now as
   the arguments to the `value_fn`. To stay future compatible, create e.g. `def value_fn(p1, pa2)`
   and using `params = ['param1, param2]`, `value_fn` will then be called as `value_fn(param1, parma2)`.

>    This will most probably break in the future.

Reviewer: ```suggestion
   `value_fn` without arguments will probably break in the future.
```
